### PR TITLE
AP_HAL_ChibiOS: remove incorrect I2C internal mask from SkystarsH7HD-bdshot

### PR DESCRIPTION
**Pull request by andyp1per**
This prevented compass detection on the external bus

Fixes https://github.com/ArduPilot/ardupilot/issues/24024

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/SkystarsH7HD-bdshot/hwdef.dat [hunk 1]
```diff
@@ -4,16 +4,14 @@
...
-undef HAL_I2C_INTERNAL_MASK DMA_PRIORITY DMA_NOSHARE
+undef DMA_PRIORITY DMA_NOSHARE
 undef DEFAULT_SERIAL6_PROTOCOL DEFAULT_SERIAL6_BAUD
 
 MCU_CLOCKRATE_MHZ 480
 
 DMA_PRIORITY TIM3* TIM4* TIM2* SPI1* SPI3* SPI4* TIM1*
 DMA_NOSHARE TIM3* TIM4* TIM2* SPI1* SPI3*
 
-define HAL_I2C_INTERNAL_MASK 2
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/SkystarsH7HD-bdshot/hwdef.dat:15]: this can change users compass orientations, are we OK with that?

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/SkystarsH7HD-bdshot/hwdef.dat:15]: should be set to 1

**Review by tridge — DISMISSED** [review]
change to define as 1, and good to merge

**Comment by mareksy12** [comment]
FW problem low brightness through osd ardupilot
https://www.youtube.com/watch?v=WiOuGGa5Nmk

**Pull request merged**